google/cloud/pubsub/samples/samples.cc

```diff
@@ -2069,6 +2065,18 @@ void AutoRun(std::vector<std::string> const& argv) {
 
   auto const snapshot_id = RandomSnapshotId(generator);
 
+  auto ignore_emulator_failures = [](auto lambda) {
+    using ::google::cloud::StatusCode;
+    try {
+      lambda();
+    } catch (google::cloud::Status const& s) {
+      if (UsingEmulator() && s.code() == StatusCode::kUnimplemented) return;
```

Comment:
Re `return` ... Might not an "unimplemented" error cascade to a non-"unimplemented" error?

Reply:
You mean like a `CreateFoo()` fails and then the next `UpdateFoo()` fails but not with `kUnimplemented`?
In this PR the existing case (in `AutoRunProtobuf()`), which already had a `if (UsingEmulator()) return;`, regardless of failures.  The PR makes things slightly less tested (we used to test 3 functions that all failed), but I can live with that.  Of course that could happen in the future, when that happens I expect that:

- When somebody introduces a PR that triggers this behavior the build will fail and they will need to do something (e.g., group all the related functions in a `AutoRunFoo()` helper so the first failure exits the function, and we just wrap the `AutoRunFoo()` call)
- If we ever changed the emulator to a new version that is also a PR and we will notice in at least some of the builds

Reply:
> You mean like a `CreateFoo()` fails and then the next `UpdateFoo()` fails but not with `kUnimplemented`?

Yes.

> When somebody introduces a PR that triggers this behavior the build will fail and they will need to do something

OK.